src/controllers/authentication/session.rs

```diff
@@ -44,16 +45,21 @@ pub async fn login(
     config: web::Data<Configuration>,
 ) -> Result<HttpResponse, actix_web::Error> {
     let username = &form.username;
-    let password = &form.password.to_string();
-    let mcrypt = &config.magic_key;
-    let encrypted_password = mcrypt.encrypt_str_to_base64(password);
+    let password = &form.password;
     let db = &config.database_connection;
-    let login_result = login_database(username, encrypted_password, db)
+
+    let parsed_hash = password_check(username.clone(), db)
         .await
         .map_err(actix_web::error::ErrorInternalServerError)?;
 
-    let logic_check_value = LoginCheck { value: 1 };
-    if login_result == logic_check_value {
+    let parsed_stored =
+        PasswordHash::new(&*parsed_hash).map_err(actix_web::error::ErrorInternalServerError)?;
+
+    let result = Argon2::default()
+        .verify_password(password.as_bytes(), parsed_stored.borrow())
+        .is_ok();
+
+    if result == true {
         Identity::login(&req.extensions(), username.to_string())
```

Comment:
Fixed it 

by             Identity::login(&req.extensions(), username.parse()?)


*username --> cannot be used because the it makes the value moved

Reply:
mismatched types [E0308] expected `String`, found `&String

if i use only username this error